clang-tools-extra/clangd/CMakeLists.txt

```diff
@@ -161,6 +163,7 @@ clang_target_link_libraries(clangDaemon
   clangAST
   clangASTMatchers
   clangBasic
+  clangDependencyScanning
```

Comment:
This dependency makes clangd depend (transitively) on clangCodeGen, which it didn't depend on previously.

This makes clangd take ~30% longer to build, increases its binary size, and it's conceptually strange that a language server contains code generation code.

Any chance we could cut the dependency on clangCodeGen again?

Reply:
It looks like it comes from https://github.com/llvm/llvm-project/blob/b634e057ddecc41dce046887d0f0854fed305374/clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp#L489-L490

but clangd didn't touch this function. So it might be possible to split that. I'll file an issue to track this.